src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -191,16 +191,18 @@ const void *ArrayBase::getBuffer() const {
 SDR *ArrayBase::getSDR() {
```

Comment:
> who handles the free()? Can you use a smart pointer?

There is no free required.  I am returning the pointer that IS in a smart pointer.  I guess I could return the entire smart pointer however.  It would complicate things a little but might work.  Originally I was returning a nullptr if the SDR is not defined but now I throw an exception which means the caller must check first.  So, returning an instance of the smart pointer is now doable.

Let me think about that for a different PR....it's in a LOT of places.

Reply:
> SDR *ArrayBase::getSDR() 

What if I return a reference rather than a pointer?
I will look at that and see how much work it is.